Gruneisen classes george

ab5424 opened the pull request:
## Summary

Test

ab5424 committed:
Initial.

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes

JaGeo committed:
Merge branch 'master' of https://github.com/materialsproject/pymatgen into gruneisen-classes-george

JaGeo committed:
First tests

JaGeo committed:
Tests Gruneisen classes

JaGeo committed:
More docu and more tests

JaGeo committed:
More docu and more tests 2

JaGeo committed:
More docu and more tests 2

JaGeo committed:
More docu and more tests 3

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes-george

JaGeo committed:
Tests and further updates

JaGeo committed:
Merge branch 'gruneisen-classes-george' of https://github.com/ab5424/pymatgen into gruneisen-classes-george

CompRhys committed:
fix: update matplotlib 3daxes to use modern api.

JaGeo committed:
Fix label bug

JaGeo committed:
Update

ab5424 committed:
Merge branch 'materialsproject:master' into gruneisen-classes-george

shyuep committed:
Merge pull request #2182 from CompRhys/3daxes

fix: update matplotlib 3daxes to use modern api.

shyuep committed:
Add check stable to get_decomp.

shyuep committed:
Merge branch 'master' of github.com:materialsproject/pymatgen

ardunn committed:
revise matproj env vars if dict read is empty

ardunn committed:
Merge branch 'master' of https://github.com/materialsproject/pymatgen

Andrew-S-Rosen committed:
Add support for spin-dependent eig band props

This commit adds support for a separate_spins keyword when calculating the eigenvalue band properties with the VASP Eigenval reader and Vasprun reader. Test cases also included.

Andrew-S-Rosen committed:
Fix no-else-return pylint error

Addresses pylint errors related to no-else-return on pymatgen.io.vasp.outputs

Andrew-S-Rosen committed:
Fix typo in test for separate_spins

Accidentally read a vasprun with Eigenval parser

Andrew-S-Rosen committed:
Add separate_spins support for BSVasprun

BSVasprun needs to have the all the arguments for eigenvalue_band_properties

Andrew-S-Rosen committed:
Add trailing comma to BSVasprun args

gpetretto commented on pymatgen/io/phonopy.py:
```diff
@@ -430,3 +440,255 @@ def get_phonon_band_structure_symm_line_from_fc(
+            structure = get_structure_from_dict(gruneisen_dict)
+        except ValueError as err:
+            print(err, "\nPlease provide a structure")
+            sys.exit(1)
```
I think it is a bad idea to to do a sys.exit here. In general it is fine when running inside a CLI, but it seems that this function would be usually called inside a script and in that case the best would be to let the exception run, so that a client can catch the exception if willing (he can also catch a SystemExit exception, but I would consider this quite dangerous)

JaGeo replied:
Fixed with a raise of a ValueError and the error message.

gpetretto commented on pymatgen/io/phonopy.py:
```diff
@@ -430,3 +440,255 @@ def get_phonon_band_structure_symm_line_from_fc(
+            structure = get_structure_from_dict(gruneisen_dict)
+        except ValueError as err:
+            print(err, "\nPlease provide a structure")
+            sys.exit(1)
```
as in the previous function, I think it best to just raise a specific error or let the exception pass

JaGeo replied:
Fixed with a raise of a ValueError and the error message.

gpetretto commented on pymatgen/io/phonopy.py:
```diff
@@ -430,3 +440,255 @@ def get_phonon_band_structure_symm_line_from_fc(
+                    d = phonon[i]['distance']
+                    distance.append(d)
+                    frequencies.append(bands)
+                    gruneisenparameters.append(gruneisenband)
```
it seems that when `fit` is True there are a few things that are not collected, e.g. `eigendisplacements`  and `phonopy_labels_dict` . Is there a reason for that?

JaGeo replied:
No. Will work on fixing this. Not sure why and at which stage this happened. Thanks for pointing this out.

JaGeo replied:
I will probably remove it. I have looked further into this: there is no option in phonopy to attach the eigendisplacements of the equillibrium structure within the gruneisen module. 

@ab5424 or did you see something here?

gpetretto commented on pymatgen/io/phonopy.py:
```diff
@@ -430,3 +440,255 @@ def get_phonon_band_structure_symm_line_from_fc(
+                            eig_b.append(eigvec_to_eigdispl(
+                                v, q, structure[i].frac_coords,
+                                structure.site_properties['phonopy_masses'][i]))
+                        eig_q.append(eig_b)
```
could this be replaced by `get_eig_b()` as well?

JaGeo replied:
Done.

Andrew-S-Rosen committed:
Update error if ISPIN = 1 and separate spins are requested

Make sure that ISPIN = 2 if separate_spins = True in eigenvalue_band_properties

Andrew-S-Rosen committed:
Modify check for ISPIN = 2 and separate_spins=True

Use .keys() instead of .items()

gpetretto commented on pymatgen/phonon/gruneisen.py:
```diff
@@ -0,0 +1,508 @@
+        b = TotalDos(a)
+        b.run()
+
+        return b
```
I am not sure what is the policy for pymatgen returning objects that are coming from other packages like in this case. 

In any case you should introduce a check that phonopy is actually installed, or the code will fail because `TotalDos` is None if phonopy is missing. You can use `@requires`

JaGeo replied:
Done.

Andrew-S-Rosen committed:
Add one more separate_spins check

Add another check for separate_spins in eigenvalue_band_properties

Andrew-S-Rosen committed:
Fix linting on eigenvalue_band_properties check

Andrew-S-Rosen committed:
Fix eigenvalue_band_properties check

Fix eigenvalue_band_properties check for separate_spins (index error)

mkhorton committed:
Merge pull request #2186 from ardunn/master

Revise #2164 - Make SETTINGS loading more robust

shyuep committed:
Update reqs

mkhorton committed:
Merge pull request #2187 from arosen93/master

Add support for spin-dependent eig band props

mkhorton committed:
Set version and update changelog

shyuep committed:
Fix README formatting.

JaGeo committed:
Upgrade

JaGeo committed:
Merge branch 'gruneisen-classes-george' of https://github.com/ab5424/pymatgen into gruneisen-classes-george

JaGeo committed:
Merge branch 'master' of https://github.com/materialsproject/pymatgen into gruneisen-classes-george

JaGeo committed:
Black reformatting

JaGeo committed:
mypy errors

JaGeo committed:
mypy errors

JaGeo committed:
Upgrade

JaGeo committed:
pydocstyle

JaGeo committed:
documetation

JaGeo committed:
reformatting

JaGeo committed:
Upgrade

JaGeo committed:
Upgrade

JaGeo committed:
Upgrade
